### DISCUSSION TITLE
Type error with createLocalizedPathnamesNavigation with pathname with dynamic params

### DISCUSSION THREAD

**Discussion by robin-ln** [amannn/next-intl#712] [Q&A]
### Description

my file `navigation.ts`

```
import { locales } from '@/configs/i18n';
import {
  Pathnames,
  createLocalizedPathnamesNavigation
} from 'next-intl/navigation';

export const pathnames = {
  '/': '/',
  '/speakers': '/speakers',
  '/speakers/[speakerId]': '/speakers/[speakerId]',
} satisfies Pathnames<typeof locales>;

export const { usePathname, useRouter, Link, redirect } =
  createLocalizedPathnamesNavigation({
    locales,
    pathnames
  });
```

but i get type error when i use Link : 

```
      <Link href="/speakers/randomSpeakersId"></Link>
```

### Mandatory reproduction URL (CodeSandbox or GitHub repository)

https://gitlab.com/robin.louarn/programme-tz-2023

### Reproduction description

Steps to reproduce:
1. replace createSharedPathnamesNavigation by createLocalizedPathnamesNavigation in src/components/navigation.ts file

### Expected behaviour

maybe use literal type for dynamique params : 
- for /speakers/[speakerId] -> get type -> `/speakers/${string}`

**Comment by amannn** [comment]
Your call site for `Link` is off, please see the docs: https://next-intl-docs.vercel.app/docs/routing/navigation#link (make sure to switch to localized pathnames).

I'll move this to a discussion since it's a usage question.
